nemo_text_processing/g2p/modules.py

```diff
@@ -579,12 +585,16 @@ def __call__(self, text):
         ' ', 'S', 't', 'o', 'r', 'e', ',', ' ', 'mai3', 'le5', 'yi2',
         'ge4', 'i', 'P', 'h', 'o', 'n', 'e', '。']
         """
-        pinyin_seq = self._lazy_pinyin(
-            text,
-            style=self._Style.TONE3,
-            neutral_tone_with_five=True,
-            errors=lambda en_words: [letter for letter in en_words],
-        )
+        pinyin_seq = []
+        # Cut sentences into words to improve polyphone disambiguation
+        words_list = list(jieba.cut(text))
+        for word in words_list:
+            pinyin_seq += self._lazy_pinyin(
+                word,
+                style=self._Style.TONE3,
```

Comment:
Should we let the user set this? (I'm not sure if the current setup does work with other representations, so if not please disregard.)

Reply:
I am not quite sure if our recipe supports new tone styles other than `TONE3`. In mandarin, `TONE3` is common. If the goal is to avoid hardcoding, we may also need to support customizing other params of func `lazy_pinyin` as shown in https://github.com/mozillazg/python-pinyin/blob/d21125f1d79449b70c60566fa2941201bb456fea/pypinyin/core.py#L327-L328

@yuekaizhang what do you think?

Reply:
Let's merge it now. We can always file a patch if there is a need to support diverse tone styles in the future. So far, the hardcoded solution is good to go.